#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 2bfafdd5c42e8f35265f7c3c6f0b91baa7d38796 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Wed, 6 Jan 2021 14:34:14 +0100
Subject: [PATCH] Allow to override kconfig executable

This makes it possible to use kconfig_add_kcfg_files when
cross compiling as it can be set in a cmake toolchain file
to have it aviable in the build systems architecture.
---
 KF5ConfigMacros.cmake | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/KF5ConfigMacros.cmake b/KF5ConfigMacros.cmake
index 7938e4c..47be213 100644
--- a/KF5ConfigMacros.cmake
+++ b/KF5ConfigMacros.cmake
@@ -89,12 +89,16 @@ function (KCONFIG_ADD_KCFG_FILES _target_or_source_var)
            file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${_rel_PATH})
        endif()

+       if (NOT EXISTS "${KF5_CONFIG_EXECUTABLE}")
+           set(KF5_CONFIG_EXECUTABLE KF5::kconfig_compiler)
+       endif()
+
        # the command for creating the source file from the kcfg file
        add_custom_command(OUTPUT ${_header_FILE} ${_src_FILE}
-                          COMMAND KF5::kconfig_compiler
+                          COMMAND ${KF5_CONFIG_EXECUTABLE}
                           ARGS ${_kcfg_FILE} ${_tmp_FILE} -d ${CMAKE_CURRENT_BINARY_DIR}/${_rel_PATH}
                           MAIN_DEPENDENCY ${_tmp_FILE}
-                          DEPENDS ${_kcfg_FILE} KF5::kconfig_compiler)
+                          DEPENDS ${_kcfg_FILE} ${KF5_CONFIG_EXECUTABLE})

        set_source_files_properties(${_header_FILE} ${_src_FILE} PROPERTIES
            SKIP_AUTOMOC ON
--
2.20.1
